TodoList/TodoList.Api/Dependency/RegisterTypes.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Net.Http;
+using System.Web;
+using Microsoft.Practices.Unity;
+using TodoList.Api.Helpers;
+using TodoList.Contracts.Api;
+using TodoList.Contracts.Dependency;
+
+namespace TodoList.Api.Dependency
+{
+    public class RegisterTypes : IRegisterTypes
+    {
+        public void RegisterType(IUnityContainer container)
+        {
+            container.RegisterType<HttpRequestMessage>(new HierarchicalLifetimeManager(), new InjectionFactory(c => RequestMessage));
```

Comment:
If you want the RequestMessage to be injected, there should be a constructor in LocationHelper that takes HttpRequestMessage. The GetLocation method can then take just the id.